### PR TITLE
DS-3088 REST API put collections / item: Added 'putCollectionsItem' i…

### PR DESCRIPTION
**Pull request by danmalthesen**
As suggested in [DS-3088](https://jira.duraspace.org/browse/DS-3088). There are already operations for CREATE and DELETE items in collections: 
POST /collections/{collectionId}/items
DELETE /collections/{collectionId}/items/{itemId} 

This adds an additional put method to the collections API
1. PUT /collections/{collectionId}/items/
2. Updates the README.md with the new method
3. Updates the RestIndex with the new method

### CODE REVIEW

**Comment by helix84** [comment]
Please, open a new pull request against the master branch. The commit seems to apply there cleanly.

DSpace 5.x won't receive any new improvements, only security and bug fixes.

**Comment by danmalthesen** [comment]
Ok, will do that instead. Thanks

File: dspace-rest/src/main/java/org/dspace/rest/CollectionsResource.java [hunk 1]
```diff
@@ -772,4 +872,59 @@ else if (!AuthorizeManager.authorizeActionBoolean(context, collection, action))
...
+				context.abort();
+				log.warn("Item(id=" + id + ") was not found!");
+				throw new WebApplicationException(Response.Status.NOT_FOUND);
+			}
+			else if (!AuthorizeManager.authorizeActionBoolean(context, item, action))
+			{
+				context.abort();
+				if (context.getCurrentUser() != null)
+				{
+					log.error("User(" + context.getCurrentUser().getEmail() + ") has not permission to "
```
Comment (DylanMeeus) [inline dspace-rest/src/main/java/org/dspace/rest/CollectionsResource.java]: Not a very 'natural' message in the English language. 
Consider: 'User (username) does not have permission to *action* this item.'
(More like the message you wrote for anonymous)

File: dspace-rest/src/main/java/org/dspace/rest/CollectionsResource.java [hunk 2]
```diff
@@ -772,4 +872,59 @@ else if (!AuthorizeManager.authorizeActionBoolean(context, collection, action))
...
+							+ getActionString(action) + " item!");
```
Comment (DylanMeeus) [inline dspace-rest/src/main/java/org/dspace/rest/CollectionsResource.java]: This does not read naturally. Consider something like the following:
"User (email) does not have permission to *action* this item".
More like how you wrote it for the 'anonymous' user

Comment (danmalthesen) [inline dspace-rest/src/main/java/org/dspace/rest/CollectionsResource.java]: I see your point, I think I might have borrowed that directly from the existing 'findCollection' method and it ended up getting a bit distorted.

**Comment by tdonohue** [comment]
As noted above, this PR needs to be recreated against the "master" branch before it can be moved forward. Flagging this as "work in progress" for now.

**Comment by tdonohue** [comment]
Thanks for your Pull Request. However, I'm closing this PR as it conflicts with our upcoming DSpace 7 REST API work (taking place currently on our `rest7` feature branch, and moving over to `master` soon). 

This PR can still move forward through one of the following routes:
1. You can choose to rework this PR to align with the new 7.x REST API efforts. Keep in mind, this will be a rebuilt REST API, and not all features have been completed/rebuilt. But, we welcome other collaborators to help move the effort forward. See https://wiki.duraspace.org/display/DSPACE/DSpace+7+UI+Working+Group
2. Or, you can resubmit this PR to our `dspace-6_x` maintenance branch as a simple patch to our 6.x REST API.

Again, thanks for the work. We hope you consider joining us in the DSpace 7 REST API efforts, but we also appreciate any patches to the existing 6.x REST API you are willing to provide.

**Pull request closed without merging**